docker/girder_worker.Dockerfile

```diff
@@ -1,3 +1,33 @@
+# ========================
+# == SERVER BUILD STAGE ==
+# ========================
+FROM python:3.7-buster as server-builder
+
+WORKDIR /home/server
+
+# https://cryptography.io/en/latest/installation/#debian-ubuntu
+RUN apt-get update
+RUN apt-get install -y build-essential libssl-dev libffi-dev python3-dev cargo
+# Recommended poetry install https://python-poetry.org/docs/#installation
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+ENV PATH="/root/.poetry/bin:$PATH"
+
+# Copy only the lock and project files
+COPY server/pyproject.toml server/poetry.lock /home/server/
+# Use the system python installation
+RUN poetry env use system
+# Skip creating a virtual env, just put stuff in /usr/local
+RUN poetry config virtualenvs.create false
+# Install dependencies only
+RUN poetry install --no-root --no-dev
+
+# Copy full source code and install again
+COPY server/ /home/server/
+RUN poetry install --no-dev
```

Comment:
Just a question here.  If you have the `.toml` and the `.lock` in the folder already with the full source code won't it just use them by default?  I'm curious as to why it requires a second install?

Reply:
It's done for the sake of build cache.  Most of the time you're changing code, not dependencies.  When you `ldc build girder` you can just skip the dependency installation and get right to linking.